### ISSUE TITLE
Cannot update logger_tls_max_lines via options

### ISSUE THREAD

**Issue by Jimjazzz** [osquery/osquery#7111]
### What operating system and version are you using?

Windows Server 2019 

host_hostname | build | platform | version
devo-ua-ft-ua-rocksdb-win-1 | 17763 | windows | 10.0.17763

### What version of osquery are you using?

4.8.0

### What steps did you take to reproduce the issue?

1. Osquery is configured to use a tls endpoing for config and logs (FleetDM). logger_tls_max_lines is not configured so it uses the value by default (1024). 
2. We are capturing windows events and the number of events is around 100 events/sec in normal state. The limit of 1024 causes the DB to grow indefinitely and other issues and we see that in every log flush the maximum is being sent.
3. We decide to increase `logger_tls_max_lines` to 8192. As we have several thousand endpoints we change the value in FleetDM options.
4. The value is returned to osquery (checked using --tls_dump)
```
  "options": {
    "config_refresh": 10,
    "disable_database": true,
    "disable_distributed": false,
    "distributed_interval": 10,
    "distributed_plugin": "tls",
    "distributed_tls_max_attempts": 3,
    "distributed_tls_read_endpoint": "/api/v1/osquery/distributed/read",
    "distributed_tls_write_endpoint": "/api/v1/osquery/distributed/write",
    "events_expiry": 300,
    "events_max": 500000,
    "logger_min_status": 1,
    "logger_plugin": "tls",
    "logger_snapshot_event_type": true,
    "logger_tls_endpoint": "/api/v1/osquery/log",
    "logger_tls_max_lines": 8192,
    "logger_tls_period": 30,
    "pack_delimiter": "/",
    "schedule_splay_percent": 10,
    "tls_session_reuse": false,
    "windows_event_channels": "System,Application,Setup,Security,Microsoft-Windows-PowerShell/Operational,ForwardedEvents"
  },
```
5. Apparently a configuration call to the TLS plugin is done (from verbose log):
`I0517 16:26:43.823305  4112 config.cpp:891] Calling configure for logger tls`

6. A query select * from osquery_flags shows that the value is changed to 8192.

### What did you expect to see?
osqueryd starts using the new max limit once the configuration is applied. Since the database has many events buffered it should start sending events near to 8192 for some flush periods until it gets normalized.

### What did you see instead?
osqueryd keeps sending 1024 events in every flush period.

### Other info
- Sometimes, I have seen osqueryd entering in a faulty state after the change where no events are logged at all, but the osqueryd process is started ( with no error logs logged) but i was not able to get any evidence for it as the behavior is not consistent. I am trying to reproduce it in verbose mode.
- Restarting the osqueryd process seems to be enough for the agent to start using the new limit.

**Comment by getvictor** [comment]
The issue is that `logger_tls_max_lines` is only applied at startup to tls logger plugin.

This PR fixes this issue by updating the tls logger configs dynamically: https://github.com/osquery/osquery/pull/8230

**Comment by getvictor** [comment]
@zwass Can you close this issue? This option can now be configured dynamically without restarting osquery as part of https://github.com/osquery/osquery/pull/8230

**Comment by directionless** [comment]
Fixed By: #8230
